src/QsCompiler/Tests.Compiler/TestCases/TypeChecking.qs

```diff
@@ -940,6 +940,25 @@ namespace Microsoft.Quantum.Testing.TypeChecking {
         let _ = Fold((x, y) -> x + y, 0, numbers);
     }
 
+    function Lambda24(xs : Int[]) : Int[] {
+        return Mapped(i -> xs[i], xs);
+    }
```

Comment:
Taking inspiration from the test cases below, one additional test case could be:
```qsharp
    function Lambda28(xs : Int[]) : Int[] {
        return Mapped(i -> xs[i+1], xs);
    }
```